### PR TITLE
Agnes to-dos

### PR DESCRIPTION
**Pull request by aagneshansson**
Things I'd like to have clarified or explained in more detail:
Command-line, github

### CODE REVIEW

File: code/src/Lib/CustomCheckbox.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    return (
+        <div>
+            <label>
+                {/* {When we check the checkbox the function on row 10 is running } */}
+                <Checkbox checked={checked} onChange={handleOnChange}></Checkbox>
+            </label>
+        </div>
+    );
+};
```
Comment (stjernberg) [inline code/src/Lib/CustomCheckbox.js:78]: Really nice job with the stylded component.👌

File: code/src/components/EmptyState.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+               </EmptyStateText>
+            );
+        };
+
+        return (
+                <EmptyStateText>
+                    What are you waiting for? Let's go!
+                </EmptyStateText>
+            );
+};
```
Comment (stjernberg) [inline code/src/components/EmptyState.js]: I like the way you divided your code into smaller components!

**Review by stjernberg — COMMENTED** [review]
Great job with the todo-app Agnes 👏 Your code look clean and is easy to follow thanks to good naming of functions and elements. You also make the components small with not too much code in each,  which makes it easier to see what each component is doing.  You've also done a great job with styled components. It seems to me that you've completed all of the blue requirements, and most of the red ones. Well done 🤩

**Review by stjernberg — APPROVED** [review]

File: code/src/Lib/Button.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+font-weight: bold;
+font-size: 16px;
+transition: all 150ms;
+cursor: pointer;
+background: #fceef5;
+
+&:hover {
+    background: #ea86b6;
+}
+`;
```
Comment (HIPPIEKICK) [inline code/src/Lib/Button.js]: Indentation of all styled components:
```
export const Button = styled.button`
  border: none;
  border-radius: 50%;
  height: 30px;
  width: 30px; 
  margin: 10px;
  justify-self: flex-end;
  font-weight: bold;
  font-size: 16px;
  transition: all 150ms;
  cursor: pointer;
  background: #fceef5;
  
  &:hover {
    background: #ea86b6;
  }
`; 
```

File: code/src/components/AddTodo.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    return (
+        <Form onSubmit={handleSubmit}>
+            <TextInput
+                type='text'
+                aria-label="Add a task"
+                value={task}
+                onChange={event => onChangeEvent(event.target.value)}
+                placeholder="Add a new task..."
+            />
```
Comment (HIPPIEKICK) [inline code/src/components/AddTodo.js]: Remember to prevent the user from submitting empty todos!

File: code/src/reducers/todos.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const initialState = {
+        list: {
+            items: [
+             ]    
+        }
+    };
+
+    export const todos = createSlice({
+    name: "todos",
+    initialState: initialState, //kan ta bort initialstate till initialstate,
```
Comment (HIPPIEKICK) [inline code/src/reducers/todos.js]: Comments in English ;)

File: code/src/reducers/todos.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    export const todos = createSlice({
+    name: "todos",
+    initialState: initialState, //kan ta bort initialstate till initialstate, 
+    
+    reducers: {
+        addTodo: (state, action) => {
+            state.list.items.push(action.payload);
+        },
+        markCompleted: (state, action) => {
```
Comment (HIPPIEKICK) [inline code/src/reducers/todos.js]: Could be named `toggleComplete`

File: code/src/reducers/todos.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            const completedTask = state.list.items.find(item => item.id === action.payload);
+            if (completedTask) {
+                completedTask.complete = !completedTask.complete;
+            }
+        },
+        removeTodo: (state, action) => {
+            state.list.items = state.list.items.filter(item => item.id !== action.payload)
+        },
+        removeAllCompleted: (state, action) => {
```
Comment (HIPPIEKICK) [inline code/src/reducers/todos.js:27]: Nice!

File: code/src/Lib/CustomCheckbox.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    border: 0;
+    clip: rect(0 0 0 0);
+    clippath: inset(50%);
+    height: 1px;
+    margin: -1px;
+    overflow: hidden;
+    padding: 0;
+    position: absolute;
+    white-space: nowrap;
+    width: 1px;
```
Comment (HIPPIEKICK) [inline code/src/Lib/CustomCheckbox.js:24]: All styled components should be indented :)

**Pull request closed without merging**